### PR TITLE
Add caching

### PR DESCRIPTION
**Pull request by patcon**
Should have a cache outside the definition, in the project root. (`resources/`?)

- [x] apps
- [ ] ~~base os~~ (need to figure out how to trick CM site, so another PR)

### CODE REVIEW

File: mia/commands/definition.py [hunk 1]
```diff
@@ -391,12 +391,15 @@ def download_apps():
...
+            os.makedirs(cache_path, mode=0o755)
+        path, http_message = urlretrieve(apk_info['package_url'], apk_path, cache_path)
         if any(http_message['status_code'] == code for code in (200, 206)):
             print('   - downloaded %s' %
                   format_file_size(http_message['Content-Length']))
         elif http_message['status_code'] == 416:
-            print('   - already downloaded. Skipping...')
+            print('   - already downloaded. Skipped.')
         else:
             raise
```
Comment (SchnWalter) [inline mia/commands/definition.py]: I just saw that one should use `raise` without a message only in `except` blocks. Please change this to something like:
```
raise Exception('   - error downloading file.')
```

File: mia/commands/definition.py [hunk 2]
```diff
@@ -391,12 +391,15 @@ def download_apps():
         if not os.path.isdir(download_path):
             os.makedirs(download_path, mode=0o755)
         apk_path = os.path.join(download_path, apk_info['package_name'])
-        path, http_message = urlretrieve(apk_info['package_url'], apk_path)
+        cache_path = os.path.join(handler.get_workspace_path(), 'resources', 'apps')
```
Comment (SchnWalter) [inline mia/commands/definition.py]: We can keep this cache_path for now, but I would change it to also include the CPU architecture in #72

**Comment by SchnWalter** [comment]
the rest is :+1:

File: mia/commands/definition.py (hunk 2 above)
Comment (patcon) [inline mia/commands/definition.py]: :+1: Good point

**Pull request merged**